cluster/aks.go

```diff
@@ -299,41 +383,45 @@ func CreateAKSClusterFromModel(clusterModel *model.ClusterModel) (*AKSCluster, e
 //AddDefaultsToUpdate adds defaults to update request
 func (c *AKSCluster) AddDefaultsToUpdate(r *components.UpdateClusterRequest) {
 
-	if r.UpdateClusterAzure == nil {
+	if r.Azure == nil {
 		log.Info("'azure' field is empty.")
-		r.UpdateClusterAzure = &banzaiAzureTypes.UpdateClusterAzure{}
+		r.Azure = &banzaiAzureTypes.UpdateClusterAzure{}
 	}
 
-	// ---- [ Node check ] ---- //
-	if r.UpdateAzureNode == nil {
-		log.Info("'node' field is empty. Load it from stored data.")
-		r.UpdateAzureNode = &banzaiAzureTypes.UpdateAzureNode{
-			AgentCount: c.modelCluster.Azure.AgentCount,
+	if r.Azure.NodePools == nil && len(r.Azure.NodePools) == 0 {
```

Comment:
Do we really need both checks? Isn't len(x) enough?

Reply:
`NodePools` field is a map and can be nil. If we just check len(x) then (if `r.Azure.NodePools` is nil) this can cause nil pointer exception

Reply:
My mistake :) `len(nil) == 0` is true